### ISSUE TITLE
How to contribute ?

### ISSUE THREAD

**Issue by pounard** [eclipse/pdt#96]
Hello,

I was attempting to setup Eclipse for contributing to PDT (long time I wrote Java, but hey, never too late to get back at it).

I followed the guide there: https://wiki.eclipse.org/PDT/Contributing#Quick_guide

It seems rather outdated, for starters:

 - http://download.eclipse.org/tools/orbit/downloads is not a valid update site anymore, at least my Eclipse tells me that it's not a valid repository,
 - https://git.eclipse.org/c/pdt/org.eclipse.pdt.git/plain/dev/psf/pdt-dev.psf is not a valid URL, it gives a 404 error.

Actually I'm stuck there, even though, a long time ago, I did wrote a few Eclipse plugins, things have changed drastically in 13 years.

Is there any up to date documentation ? Or did I misdone something ?

Many thanks in advance for the help.

**Comment by pounard** [comment]
OK ! Found the right psf links there: https://projects.eclipse.org/projects/tools.pdt/developer : https://raw.githubusercontent.com/eclipse/pdt/master/dev/psf/pdt-dev.psf

I'll continue and come back on this issue if I encounter any other problem. Sorry for the noise !

**Comment by pounard** [comment]
OK, did that: "(Optional) Disable API Baselines errors (Window → Preferences menu item, then Plug-in Development → API Baselines, then choose Missing API baseline → Ignore)"

Now I have only 3576 problems :)

**Comment by pounard** [comment]
Pfiou, that was really a long time ago I worked on Eclipse plugins, after the first run, I no longer have all those errors, only one remaining. I forgot about that !

Good job, it works !

Still got one problem thought:

```
Unknown extension point: 'org.eclipse.wildwebdeveloper.json.schema'	plugin.xml	/org.eclipse.php.composer.core	line 157	Plug-in Problem
```

I guess the quick startup is missing a few dependencies to install, I'll try to poke things around and see if I manage to find my way in PDT.

**Comment by mlocati** [comment]
@pounard you can update the contributing wiki page yourself

**Comment by pounard** [comment]
I can do that ?

**Comment by mlocati** [comment]
I don't remember, maybe you first need to login, then you should see an Edit button somewhere

**Comment by pounard** [comment]
About contributing to the wiki page, I'm not sure of myself enough right now. For example, regarding the Orbit dependency update site that's wrong, how I could find the right one ? How could I tell if it's still required or not ? My guess is that it might be not, since I managed to run it.

**Comment by pounard** [comment]
Fun continues, I cannot run tests:

```
Required plug-in 'org.eclipse.jdt.junit.runtime' could not be found.
```

I'm searching the web for help about this, I can't find anything related to JUnit in the software updater.

**Comment by zulus** [comment]
My simple solution:

1. Install Eclipse for Committers: https://www.eclipse.org/downloads/packages/release/2020-09/r/eclipse-ide-eclipse-committers
2. Install DLTK and PDT: https://download.eclipse.org/tools/pdt/updates/latest-nightly/ and https://download.eclipse.org/technology/dltk/updates-dev/latest-nightly/
3. Import maven projects

**Comment by pounard** [comment]
@zulus exactly what I ended up with ! But I still get errors related to junit.

**Comment by ibakirov** [comment]
@pounard @zulus I agree wiki page and README file should be updated to how correctly checkout project structure and start hacking.

**Comment by mhoffrog** [comment]
@ibakirov @pounard @zulus I did create pull request #136 to leverage the Maven import procedure getting a proper Eclipse IDE set up. Please review the pull request. After that one the Wiki page can be updated accordingly. The steps of the procedure are described by the pull request comment.

**Comment by pounard** [comment]
@mhoffrog very nice ! Thanks, I'll try when I can, in the meantime when I opened this issue, two kids happened, so time is something I don't have much anymore...

**Comment by ibakirov** [comment]
@mhoffrog Thanks for your effort, I will try to follow pull request and by the way I rarely use Eclipse
